projects/onto-search/src/lib/onto-search-facet/datepicker-facet/datepicker-facet.component.ts

```diff
@@ -0,0 +1,182 @@
+import {Component, EventEmitter, Input, Output, Renderer2, TemplateRef} from '@angular/core';
+import {OntoSearchFacetComponent} from '../onto-search-facet.component';
+import {FormControl, FormGroup} from '@angular/forms';
+import {SearchDateFacetGroupModel} from './models/search-date-facet-group-model';
+import moment, {Moment} from 'moment';
+import {DateRange} from '@angular/material/datepicker';
+import {SearchDateFacetModel} from './models/search-date-facet-model';
+import {SearchDateFacetRange} from './models/search-date-facet-range';
+
+@Component({
+  selector: 'onto-datepicker-facet',
+  templateUrl: './datepicker-facet.component.html',
+  styleUrls: ['./datepicker-facet.component.scss']
+})
+export class DatepickerFacetComponent extends OntoSearchFacetComponent {
+  @Input()
+  public data: SearchDateFacetGroupModel;
+
+  @Input()
+  public facetTemplate: TemplateRef<any>;
+
+  @Input()
+  public facetTitleTemplate: TemplateRef<any>;
+
+  @Output()
+  public onSelectionChange: EventEmitter<SearchDateFacetRange> = new EventEmitter<SearchDateFacetRange>();
+
+  dateRangeGroup: FormGroup = new FormGroup({
+    start: new FormControl(),
+    end: new FormControl()
+  });
+
+  datePickerPlaceholder: string;
+
+  private daysMap: Map<string, number> = new Map<string, number>();
+  private monthsMap: Map<string, number> = new Map<string, number>();
+  private yearsMap: Map<string, number> = new Map<string, number>();
+
+  private buttonUnsubscribeFns: (() => void)[] = [];
+
+  constructor(private renderer: Renderer2) {
+    super();
+  }
+
+  ngOnInit(): void {
+    super.ngOnInit();
+    this.mapFacetDates(this.data.facetGroupData);
+
+    this.datePickerPlaceholder = this.data.placeholder;
+    if (this.data.selectedRange) {
+      this.setSelectedRange(this.data.selectedRange);
+    }
+  }
+
+  calendarOpened(event): void {
+    setTimeout(() => {
+      this.buttonUnsubscribeFns.forEach((unsubscribeFn) => {
+        unsubscribeFn();
+      });
+
+      this.buttonUnsubscribeFns = [];
+      const buttons = document.querySelectorAll('mat-calendar .mat-calendar-body-cell, mat-calendar button, mat-calendar .mat-icon-button');
+      buttons.forEach((btn) => {
+        const unsubscribe = this.renderer.listen(btn, 'click', () => {
```

Comment:
this should be unsubscribed on destroy as well

Also move the subscribtion and unsubscription in a seperate methods for better readability and reusability